fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,426 @@
+     * state machine are as follows:
+     *
+     * <p>New -> Scheduled:
+     *
+     * <p>-- When the lake table is newly created, do: schedule a timer to wait for a tiering
+     * interval configured in table, transmit the table to Pending
+     *
+     * <p>ReInitialized -> Scheduled：
+     *
+     * <p>-- When the coordinator server is restarted, for all existing lake table, do: schedule a
+     * timer to wait for a tiering service timeout, transmit the tables to Pending. Note, we must
+     * wait for the tiering service timeout in case any existing tiering service is tiering the
+     * table
+     *
+     * <p>Tiering -> Scheduled
+     *
+     * <p>-- When the tiering service finished one-round of tiering, do: schedule a timer to wait
+     * for a tiering interval configured in table, transmit the table to Pending
+     *
+     * <p>Scheduled -> Pending
+     *
+     * <p>-- The time interval to wait has passed, do: transmit to Pending state
+     *
+     * <p>Tiering -> Pending
+     *
+     * <p>-- When the tiering service is timeout, do: transmit to Pending state
+     *
+     * <p>Pending -> Tiering
+     *
+     * <p>-- When the table is assigned to a tiering service after tiering service request the
+     * table, do: transmit to Tiering state
+     */
+    private void doHandleStateChange(long tableId, TieringState targetState) {
+        TieringState currentState = tieringStates.get(tableId);
+        if (!isValidStateTransition(currentState, targetState)) {
+            LOG.error(
+                    "Fail to change state for table {} from {} to {} as it's not a valid state change.",
+                    tableId,
+                    currentState,
+                    targetState);
+            return;
+        }
+        switch (targetState) {
+            case New:
+            case ReInitialized:
+                // do nothing
+                break;
+            case Scheduled:
+                if (currentState == TieringState.New) {
+                    scheduleTableTiering(tableId);
+                } else if (currentState == TieringState.ReInitialized) {
+                    // schedule the table, but we wait for a period of TIMEOUT
+                    // to avoid the existing tiering services are still tiering it since
+                    // it's a table reinitialized by coordinator server
+                    scheduleTableTiering(tableId, TIERING_SERVICE_TIMEOUT_MS);
+                } else if (currentState == TieringState.Tiering) {
+                    // should happens the table has finished one-round of tiering,
+                    // schedule it again
+                    liveTieringTableIds.remove(tableId);
+                    scheduleTableTiering(tableId);
+                }
+                break;
+            case Pending:
+                pendingTieringTables.add(tableId);
+                break;
+            case Tiering:
+                // do nothing
```

Comment:
As I commented on `registerTieringTimeout` method, we should add initial heartbeat (current system time) to `liveTieringTableIds`, and let the timer thread scan the `liveTieringTableIds` to check the timeout tables periodically.